src/integer/mat_poly_over_z/arithmetic/mul_scalar.rs

```diff
@@ -0,0 +1,286 @@
+        }
+        out
+    }
+}
+
+arithmetic_trait_reverse!(Mul, mul, PolyOverZ, MatPolyOverZ, MatPolyOverZ);
+
+arithmetic_trait_borrowed_to_owned!(Mul, mul, MatPolyOverZ, PolyOverZ, MatPolyOverZ);
+arithmetic_trait_borrowed_to_owned!(Mul, mul, PolyOverZ, MatPolyOverZ, MatPolyOverZ);
+arithmetic_trait_mixed_borrowed_owned!(Mul, mul, MatPolyOverZ, PolyOverZ, MatPolyOverZ);
+arithmetic_trait_mixed_borrowed_owned!(Mul, mul, PolyOverZ, MatPolyOverZ, MatPolyOverZ);
+
+#[cfg(test)]
+mod test_mul_z {
+
+    use crate::integer::MatPolyOverZ;
+    use crate::integer::Z;
+    use std::str::FromStr;
+
+    /// Checks if matrix multiplication works fine for both borrowed
+    #[test]
+    fn borrowed_correctness() {
+        let mat1 = MatPolyOverZ::from_str("[[2  1 42,1  17],[1  8,2  1 2]]").unwrap();
+        let mat2 = mat1.clone();
+        let mat3 = MatPolyOverZ::from_str("[[2  2 84,1  34],[1  16,2  2 4]]").unwrap();
+        let integer = Z::from(2);
+
+        let mat1 = &mat1 * &integer;
+        let mat2 = &integer * &mat2;
+
+        assert_eq!(mat3, mat1);
+        assert_eq!(mat3, mat2);
+    }
+
+    /// Checks if scalar multiplication works fine for both owned
+    #[test]
+    fn owned_correctness() {
+        let mat1 = MatPolyOverZ::from_str("[[2  1 42,1  17],[1  8,2  1 2]]").unwrap();
+        let mat2 = mat1.clone();
+        let mat3 = MatPolyOverZ::from_str("[[2  2 84,1  34],[1  16,2  2 4]]").unwrap();
+        let integer1 = Z::from(2);
+        let integer2 = Z::from(2);
+
+        let mat1 = mat1 * integer1;
+        let mat2 = integer2 * mat2;
+
+        assert_eq!(mat3, mat1);
+        assert_eq!(mat3, mat2);
+    }
+
+    /// Checks if scalar multiplication works fine for half owned/borrowed
+    #[test]
+    fn half_correctness() {
+        let mat1 = MatPolyOverZ::from_str("[[2  1 42,1  17],[1  8,2  1 2]]").unwrap();
+        let mat2 = mat1.clone();
+        let mat3 = mat1.clone();
+        let mat4 = mat1.clone();
+        let mat5 = MatPolyOverZ::from_str("[[2  2 84,1  34],[1  16,2  2 4]]").unwrap();
+        let integer1 = Z::from(2);
+        let integer2 = Z::from(2);
+
+        let mat1 = mat1 * &integer1;
+        let mat2 = &integer2 * mat2;
+        let mat3 = &mat3 * integer1;
+        let mat4 = integer2 * &mat4;
+
+        assert_eq!(mat5, mat1);
+        assert_eq!(mat5, mat2);
+        assert_eq!(mat5, mat3);
+        assert_eq!(mat5, mat4);
+    }
+
+    /// Checks if scalar multiplication works fine for different scalar types
+    #[test]
+    fn different_types() {
+        let mat1 = MatPolyOverZ::from_str("[[1  42],[0],[2  1 2]]").unwrap();
+        let mat2 = MatPolyOverZ::from_str("[[1  2,1  6,1  5],[1  4,2  17 42,1  3]]").unwrap();
+        let mat3 = MatPolyOverZ::from_str("[[1  84],[0],[2  2 4]]").unwrap();
+        let mat4 = MatPolyOverZ::from_str("[[1  4,1  12,1  10],[1  8,2  34 84,1  6]]").unwrap();
+
+        assert_eq!(mat3, 2 * mat1);
```

Comment:
If you want to further optimize: check also for `0 * mat1` and `-1 * mat1` to check for negative and zero values. But these changes are highly optional